transformer_engine/pytorch/module/linear.py

```diff
@@ -240,7 +240,8 @@ def forward(
         weightmat = weight
         if fp8 or debug:
             # Configure quantizer
-            if weight_quantizer is not None:
+            # No need to set the quantizer states if weight is already quantized
+            if weight_quantizer is not None and not isinstance(weight, QuantizedTensor):
```

Comment:
We want `get_weight_workspace` to actually respect the quantizer that's passed into it. Otherwise, you need to know its internal details to reason out the cases where it follows the quantizer and the cases where it ignores it. We should add logic so that `columnwise_usage=False` if FSDP is enabled.

Reply:
Hm, I see your argument. If the weights are quantized then the damage is already done. No benefit trying to fix usages here, and we might as well update usages lazily. I think your approach is reasonable.

I'll just nit the comment below ("In the case of FSDP2...") since this is not specific to FSDP, but for any quantized weights. The comment above ("No need to set the quantizer states...") is more accurate.